src/client/datascience/jupyter/kernels/cellExecution.ts

```diff
@@ -131,26 +136,29 @@ export class CellExecution {
         }
 
         this.initPromise = this.enqueue();
+        workspace.onDidCloseTextDocument(
```

Comment:
Clean kernels.
If user deletes a cell, then we need to dispose related code & references.
We should also cancel the execution.